utilities/fault_injection_fs.cc

```diff
@@ -485,26 +485,36 @@ IOStatus TestFSRandomAccessFile::ReadAsync(
     FSReadRequest& req, const IOOptions& opts,
     std::function<void(FSReadRequest&, void*)> cb, void* cb_arg,
     void** io_handle, IOHandleDeleter* del_fn, IODebugContext* /*dbg*/) {
-  IOStatus s;
+  IOStatus res_status;
   FSReadRequest res;
+  IOStatus s;
   if (!fs_->IsFilesystemActive()) {
-    s = fs_->GetError();
+    res_status = fs_->GetError();
   }
-  if (s.ok()) {
-    s = fs_->MaybeInjectThreadLocalError(
+  if (res_status.ok()) {
+    res_status = fs_->MaybeInjectThreadLocalError(
         FaultInjectionIOType::kRead, opts,
         FaultInjectionTestFS::ErrorOperation::kRead, &res.result,
         use_direct_io(), req.scratch, /*need_count_increase=*/true,
         /*fault_injected=*/nullptr);
   }
-  if (s.ok()) {
+  if (res_status.ok()) {
     s = target_->ReadAsync(req, opts, cb, cb_arg, io_handle, del_fn, nullptr);
     // TODO (low priority): fs_->ReadUnsyncedData()
   }
-  if (!s.ok()) {
-    res.status = s;
+  // If there’s no injected error, then cb will be called asynchronously when
+  // target_ actually finishes the read. But if there’s an injected error, it
+  // needs to immediately call cb(res, cb_arg) s since target_->ReadAsync()
+  // isn’t invoked at all.
+  if (!res_status.ok()) {
```

Comment:
nit: To me, it would be more readable under `else` in line 504, since we are doing the following `if` no injected error (`res_status.ok()`), let cb be called asynchronously after target_ finishes. Otherwise (`else`), call `cb` immediately.

Reply:
sounds good will fix

Reply:
Fixed